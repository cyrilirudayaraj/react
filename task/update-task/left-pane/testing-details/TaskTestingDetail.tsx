import React from 'react';
import { Form } from '@athena/forge';
import Labels from '../../../../../constants/Labels';
import { TaskTestingDetailProps } from '../../UpdateTaskProps';
import { isTaskEditable } from '../../UpdateTask';
import Acl from '../../../../../constants/Acl';
import WFEnableForPermission from '../../../../../components/wf-enableforpermission/WFEnableForPermission';
import RichTextEditor from '../../../../../components/rich-text-editor/RichTextEditor';

export default function TaskTestingDetail(
  props: TaskTestingDetailProps
): JSX.Element {
  const labels = Labels.TASK_TESTING_DETAILS;

  return (
    <Form
      labelAlwaysAbove={true}
      includeSubmitButton={false}
      className="task-detail"
    >
      <WFEnableForPermission permission={Acl.TASK_TEST_NOTES_UPDATE}>
        <RichTextEditor
          id="testNote"
          wrapperId={3}
          labelText={labels.TEST_NOTE}
          formik={props.formik}
          isDisabled={!isTaskEditable(props.task)}
          isRequired={props?.mandatoryFields['testNote'] || false}
        ></RichTextEditor>
      </WFEnableForPermission>
    </Form>
  );
}
