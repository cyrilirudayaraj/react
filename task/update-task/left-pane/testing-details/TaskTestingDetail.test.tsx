import { configure, shallow } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import TaskTestingDetail from './TaskTestingDetail';

configure({ adapter: new Adapter() });
import { FormikValues } from 'formik';

describe('TaskTestingDetail', () => {
  const props: FormikValues = {
    values: {},
    touched: {},
    errors: {},
    handleChange: (a) => a,
    handleBlur: (a) => a,
    handleSubmit: (a) => a,
    getFormFieldPRops: jest.fn(),
    getFieldProps: jest.fn(),
  };

  const testDetailComponent = shallow(
    <TaskTestingDetail mandatoryFields={jest.fn()} formik={props} />
  );
  it('should render as expected', () => {
    expect(testDetailComponent.debug()).toMatchSnapshot();
  });
});
