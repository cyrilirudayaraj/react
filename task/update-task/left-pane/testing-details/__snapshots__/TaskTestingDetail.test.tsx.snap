// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTestingDetail should render as expected 1`] = `
"<Form labelAlwaysAbove={true} includeSubmitButton={false} className=\\"task-detail\\" buttonText=\\"Submit\\" noValidate={true} requiredVariation=\\"blueBarWithRequiredLabel\\" layout=\\"medium\\" nested={true} labelWidth={3}>
  <Connect(WFEnableForPermission) permission=\\"task.test-notes.update\\">
    <Connect(RichTextEditor) id=\\"testNote\\" wrapperId={3} labelText=\\"Test Note\\" formik={{...}} isDisabled={false} isRequired={false} />
  </Connect(WFEnableForPermission)>
</Form>"
`;
